R/digest.R

```diff
@@ -0,0 +1,39 @@
+# The functions here support making digests stable when there are no model changes.
+
+#' @export
+sha1.rxUi <- function(x, digits = 14L, zapsmall = 7L, ..., dropSessionSpecific = TRUE, algo = "sha1") {
+  object <- rxUiDecompress(x)
+  nmCheck <- names(object)
+  if (dropSessionSpecific) {
+    nmCheck <- setdiff(nmCheck, "modelName")
+  }
+  objectSha1 <- character()
+  for (nm in nmCheck) {
+    objectSha1[[nm]] <- sha1(object[[nm]], digits = digits, zapsmall = zapsmall, ..., dropSessionSpecific = dropSessionSpecific, algo = algo)
+  }
+  sha1(objectSha1, digits = digits, zapsmall = zapsmall, ..., dropSessionSpecific = dropSessionSpecific, algo = algo)
+}
+
+#' @export
+sha1.rxModelVars <- function(x, digits = 14L, zapsmall = 7L, ..., dropSessionSpecific = TRUE, algo = "sha1") {
+  nmCheck <- names(x)
+  if (dropSessionSpecific) {
```

Comment:
@mattfidler, These object parts are some of the places where I have sometimes seen differences between the same model, and sometimes I don't.

Reply:
Hi @billdenney 

This is contained in the model variables already. If it isn't working there, we need to refactor it.  It has implications on compiling and cashing.  I think this method shouldn't not be added (as written).